src/irmin/metrics.mli

```diff
@@ -14,65 +14,70 @@
+    is to decouple the metrics type definition from the data manipulation.
+
+    It has a 'a {!repr} type to describe the type of the metric manipulated.
+    This {!repr} is used to define a {!metric}. It is in charge of how it is
+    gathered and how it should be stored (if needed).
+
+    A {!metric} can be modified in different ways, depending on the {!modifier}. *)
 
 type origin = ..
 (** An extensible type to get the location of the definition. *)
 
-type origin += Default  (** Default {!origin} for the {!data} definition. *)
-
-type 'a metric
-(** {!metric} caracterizes the type of the metric that is manipulated. *)
+type 'a repr
+(** {!ty} is the {!Repr.ty} description of the type used by metrics. *)
 
-type ('a, 'b) data
-(** {!data} is the object that describes how a {!metric} is gathered and store. *)
+val repr : string -> 'a Repr.ty -> 'a repr
+(** [repr name t] generates a metric representation, named according to [name]. *)
 
-val value : ('a, 'b) data -> 'a
-(** The value field of a {!data} record. *)
+type ('a, 'b) metric
+(** {!metric} is the object that describes how a {!metric} is gathered and
+    store. *)
 
-val set_value : ('a, 'b) data -> 'a -> unit
-(** [set_value data v] updates the value in the {!data} object. *)
+val value : ('a, 'b) metric -> 'a
+(** The value extracted from a {!metric}. *)
 
-val storage : ('a, 'b) data -> 'b
-(** The storage field of a {!data} record. *)
+val set_value : ('a, 'b) metric -> 'a -> unit
+(** [set_value m v] updates the value in the {!metric} object. *)
 
-val set_storage : ('a, 'b) data -> 'b -> unit
-(** [set_storage data s] update the storage int the {!data} object. *)
+val storage : ('a, 'b) metric -> 'b
+(** The storage extracted from a {!metric}. *)
 
-(** {!kind} describes how the data will be handled by the {!produce} function.
+val set_storage : ('a, 'b) metric -> 'b -> unit
+(** [set_storage m s] updates the storage in the {!metric} object. *)
 
-    - Ref: the value and the storage are not modified
-    - Value f: apply f to the value
-    - Storage f: apply f to the storage and the value
+(** {!modifier} describes how the data will be handled by the {!produce}
+    function.
 
-    It gives the possibility to have the same data manipulate in different ways. *)
-type ('a, 'b) kind = Ref | Value of ('a -> 'a) | Storage of ('a -> 'b -> 'b)
+    - Mutable: the value and the storage are not modified but the content of the
+      value can be mutate.
+    - Value f: apply f to the value and updates its content.
+    - Mem f: apply f to the storage and the value and updates the storage
+      content.
 
-val define : string -> 'a Repr.t -> 'a metric
-(** [define name ty] creates a {ty metric}. *)
+    It gives the possibility to handle the same metric in different ways. *)
+type ('a, 'b) modifier =
+  | Mutable of ('a -> unit)
+  | Value of ('a -> 'a)
+  | Mem of ('a -> 'b -> 'b)
 
-val init :
-  'a metric ->
+val create :
```

Comment:
`v` ? :p (it's short, but it's kind of consistent with the rest of the create function in the Irmin codebase)